akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSpec.scala

```diff
@@ -48,71 +58,22 @@ object FlowSpec {
     }
   }
 
-  class BrokenActorFlowMaterializer(
-    settings: ActorFlowMaterializerSettings,
-    dispatchers: Dispatchers,
-    supervisor: ActorRef,
-    flowNameCounter: AtomicLong,
-    namePrefix: String,
-    brokenMessage: Any) extends ActorFlowMaterializerImpl(settings, dispatchers, supervisor, flowNameCounter, namePrefix) {
-
-    override def processorForNode[In, Out](op: AstNode, flowName: String, n: Int): (Processor[In, Out], MaterializedMap) = {
-      val props = op match {
-        case f: Fused            ⇒ Props(new BrokenActorInterpreter(settings, f.ops, brokenMessage))
-        case Map(f, att)         ⇒ Props(new BrokenActorInterpreter(settings, List(fusing.Map(f, att.settings(settings).supervisionDecider)), brokenMessage))
-        case Filter(p, att)      ⇒ Props(new BrokenActorInterpreter(settings, List(fusing.Filter(p, att.settings(settings).supervisionDecider)), brokenMessage))
-        case Drop(n, _)          ⇒ Props(new BrokenActorInterpreter(settings, List(fusing.Drop(n)), brokenMessage))
-        case Take(n, _)          ⇒ Props(new BrokenActorInterpreter(settings, List(fusing.Take(n)), brokenMessage))
-        case Collect(pf, att)    ⇒ Props(new BrokenActorInterpreter(settings, List(fusing.Collect(att.settings(settings).supervisionDecider)(pf)), brokenMessage))
-        case Scan(z, f, att)     ⇒ Props(new BrokenActorInterpreter(settings, List(fusing.Scan(z, f, att.settings(settings).supervisionDecider)), brokenMessage))
-        case Expand(s, f, _)     ⇒ Props(new BrokenActorInterpreter(settings, List(fusing.Expand(s, f)), brokenMessage))
-        case Conflate(s, f, att) ⇒ Props(new BrokenActorInterpreter(settings, List(fusing.Conflate(s, f, att.settings(settings).supervisionDecider)), brokenMessage))
-        case Buffer(n, s, _)     ⇒ Props(new BrokenActorInterpreter(settings, List(fusing.Buffer(n, s)), brokenMessage))
-        case MapConcat(f, att)   ⇒ Props(new BrokenActorInterpreter(settings, List(fusing.MapConcat(f, att.settings(settings).supervisionDecider)), brokenMessage))
-        case o                   ⇒ ActorProcessorFactory.props(this, o)
-      }
-      val impl = actorOf(props.withDispatcher(settings.dispatcher), s"$flowName-$n-${op.attributes.name}")
-      (ActorProcessorFactory(impl), MaterializedMap.empty)
-    }
-
+  val faultyFlow: Flow[Any, Any, _] ⇒ Flow[Any, Any, _] = in ⇒ in.andThenMat { () ⇒
+    val props = Props(new BrokenActorInterpreter(settings, List(fusing.Map({ x: Any ⇒ x }, stoppingDecider)), "a3"))
+      .withDispatcher("akka.test.stream-dispatcher")
+    val processor = ActorProcessorFactory[Any, Any](system.actorOf(
```

Comment:
yes, I am attaching the broken processor to the system. we can probably fix this.